x-pack/plugin/otel-data/src/main/resources/component-templates/traces-otel@mappings.yaml

```diff
@@ -0,0 +1,65 @@
+---
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for OpenTelemetry traces
+  managed: true
+template:
+  mappings:
+    date_detection: false
+    properties:
+      trace_id:
+        type: keyword
+      trace.id:
+        type: alias
+        path: trace_id
+      span_id:
+        type: keyword
+      span.id:
+        type: alias
+        path: span_id
+      trace_state:
+        type: keyword
+      parent_span_id:
+        type: keyword
+      parent.id:
+        type: alias
+        path: parent_span_id
+      trace_flags:
+        type: byte
+      name:
+        type: keyword
+      span.name:
+        type: alias
+        path: name
+      kind:
+        type: keyword
+      duration:
+        type: long
+        meta:
+          unit: nanos
+      dropped_events_count:
+        type: long
+      links:
```

Comment:
Q: does this handle array of objects properly? Is it already implicitly defined as an object type?

Reply:
> we probably need to set store_array_source so that synthetic source can re-construct in the correct order

quote from @felixbarny

Reply:
This flag isn't documented, yet but you can read more about it in the PR: https://github.com/elastic/elasticsearch/pull/108417

Reply:
Pushed it in https://github.com/elastic/elasticsearch/pull/111091/commits/846db1d13759a5d4edff0ec8647f81969bc54d07 including a test.